src/sdk/glacier.js

```diff
@@ -154,13 +179,35 @@ class GlacierBackend {
      * @returns {nb.RestoreStatus | undefined}
      */
     static get_restore_status(xattr, now, file_path) {
-        if (xattr[GlacierBackend.STORAGE_CLASS_XATTR] !== s3_utils.STORAGE_CLASS_GLACIER) return;
+        const storage_class = Glacier.storage_class_from_xattr(xattr);
+        if (storage_class !== s3_utils.STORAGE_CLASS_GLACIER) {
+            return;
+        }
+
+        const tape_info = Glacier.parse_tape_info(xattr);
+
+        if (Glacier.is_externally_managed(xattr)) {
```

Comment:
I think we should have the option to set restore status from dmapi even when IMPLICIT_SC is not enabled.